ui/web/index.html

```diff
@@ -3,10 +3,323 @@
+                </tr>
+                <tr>
+                    <td>
+                        <div id="chartdiv" style="height:500px;width:300px; "></div>
+                    </td>
+                </tr>
+                </table>
+    </fieldset>
+
+    <script>  
+
+        var columns_mapping = {};
+        var requestBody = {
+            columns_mapping
+        };
+        var applicationContext = {};
+        var itemNumber;
+        var tasks = [];
+        var inquiryPoints = {"numberOfPoints": 50, 
+        tasks
+        };
+
+        var tasksObject = {};
+        var tasksPlot = {
+            tasksObject
+        };
+                                   
+        $("#loadTasksButton").click(dataInput); 
+            function dataInput() {
+
+            var excelFile = $('#excelFile').get(0).files.item(0);
+            requestBody.sheet = $("#sheetName").val();
+            requestBody.first_row = +$("#firstRow").val();
+            requestBody.last_row = +$("#lastRow").val();
+            requestBody.rows_to_skip = $("#rowSkip").val();
+            columns_mapping.uid = $("#uid").val();
+            columns_mapping.name = $("#nam").val();
+            columns_mapping.blockers = $("#blockers").val();
+            columns_mapping.min_estimate = $("#minEst").val();
+            columns_mapping.normal_estimate = $("#norEst").val();
+            columns_mapping.max_estimate = $("#maxEst").val(); 
+            $.ajax({
+                type: 'PUT',
+                url: 'http://localhost:5000/api/excelWorkbook',
+                data: excelFile,
+                contentType: 'application/vnd.ms-excel',
+                processData: false,
+                success: function (data) {
+                    alert("Response received: " + JSON.stringify(data));
+                    requestBody.file = data.tempFileName;
+                    $.ajax({
+                        type: 'POST',
+                        url: 'http://localhost:5000/api/excelWorkbook/tasks',
+                        contentType: 'application/json',
+                        data: JSON.stringify(requestBody),
+                        success: function (data) {
+                            alert("Response received: " + JSON.stringify(data));
+                            applicationContext.tasks = data.tasks;
+                        }    
+                    });    
+                }
+             });                           
+        };
+
+        $("#loadPrint").click(dataOutput);
+            function dataOutput() {
+
+            var numTask = applicationContext['tasks'];
+            var tasksNumber = $("#tasks").val();
+     
+            for(var i = 0, iLen = numTask.length; i < iLen; i++){
+                if(i == +tasksNumber){
+                    itemNumber = numTask[i];
+                    delete itemNumber.blockers;
+                    delete itemNumber.name;
+                    tasks.push(itemNumber);
+                }
+            }
```

Comment:
Code above looks confusing. I would suggest rewriting of this logic in next way:

```
var taskToPrintIndex = $("#tasks").val(); 
var taskToPrint = applicationContext.tasks[taskToPrintIndex]

var inquiryPoints = {
  "numberOfPoints": 50, 
   "tasks": [taskToPrint]
};
```

Reply:
Recommended changes made